### PR TITLE
 Upload: Turn on S3 Transfer Acceleration by default

### PR DESCRIPTION
**Pull request by sampierson**
and provide a --no-transfer-acceleration option to turn it off.

### CODE REVIEW

File: hca/upload/__init__.py [hunk 1]
```diff
@@ -52,14 +52,19 @@ def list_areas():
     return [{'uuid': area.uuid, 'is_selected': area.is_selected} for area in UploadArea.all()]
 
 
-def upload_file(file_path, target_filename=None, report_progress=False, dcp_type="data"):
+def upload_file(file_path, target_filename=None, use_transfer_acceleration=True, report_progress=False,
```
Comment (ttung) [inline hca/upload/__init__.py]: I recommend using the one-parameter-per-line style when wrapping like this.

File: hca/upload/cli/upload_command.py [hunk 2]
```diff
@@ -23,20 +24,30 @@ def add_parser(cls, upload_subparsers):
         upload_parser.add_argument('-t', '--target-filename', metavar="<filename>", default=None,
                                    help="Filename to use in upload area (if you wish to change it during upload)." +
                                    " Only valid when one file is being uploaded.")
+        upload_parser.add_argument('--no-transfer-acceleration', action='store_true',
+                                   help="""Don't use Amazon S3 Transfer Acceleration.
+                                           By default we using the aforementioned service to upload via an endpoint
+                                           geographically close to you, instead of directly to Virginia, USA.
+                                           However in some situations this can be slower.  If it is, add this option.
```
Comment (ttung) [inline hca/upload/cli/upload_command.py]: nit: `However, in`

File: hca/upload/cli/upload_command.py [hunk 3]
```diff
@@ -23,20 +24,30 @@ def add_parser(cls, upload_subparsers):
...
+                                           Google "S3 Transfer Acceleration Speed Comparison Tool" for details.""")
```
Comment (ttung) [inline hca/upload/cli/upload_command.py]: Maybe just drop a direct link?

File: test/upload/cli/test_upload.py [hunk 4]
```diff
@@ -45,7 +45,12 @@ def test_upload_with_target_filename_option(self):
         s3 = boto3.resource('s3')
         s3.Bucket(TEST_UPLOAD_BUCKET).create()
 
-        UploadCommand(Namespace(file_paths=['test/bundle/sample.json'], target_filename='FOO', quiet=True))
+        args = Namespace(
```
Comment (ttung) [inline test/upload/cli/test_upload.py]: Is there value in testing once with no_transfer_acceleration=True to verify if the transfer config is set up correctly?

Comment (sampierson) [reply test/upload/cli/test_upload.py]: Done

**Pull request merged**